pymongo/pool.py

```diff
@@ -307,6 +337,13 @@ def _getenv_int(key: str) -> Optional[int]:
 
 def _metadata_env() -> dict[str, Any]:
     env: dict[str, Any] = {}
+    container = get_container_env_info()
+    if container is not None:
+        env["container"] = {}
+        if container.runtime:
+            env["container"]["runtime"] = container.runtime
+        if container.orchestrator:
+            env["container"]["orchestrator"] = container.orchestrator
```

Comment:
What do you think about inlining the get_container_env_info() logic and removing the ContainerInfo class? I believe it will make this code more straightforward to read.

Reply:
I like the extensibility of having a separate `get_container_env_info` function + `ContainerInfo` class. If we add similar detection for other platforms, or want more verbose metadata for K8s or Docker, inlining will no longer be very readable.

Reply:
I think it's the opposite. The inlined version removes 20~ lines of code and is much easier to read:
```python
    container = {}
    if Path(DOCKER_ENV_PATH).exists():
        container["runtime"] = RUNTIME_NAME_DOCKER
    if os.getenv(ENV_VAR_K8S):
        container["orchestrator"] = ORCHESTRATOR_NAME_K8S
    if container:
        env["container"] = container
```

Reply:
It does now, but if we add further metadata like this inline we'll want to move it out for readability. I'm fine going with the inline version now and coming back to it if we add more in the future.

Reply:
What about a compromise where we just remove the dataclass and return the "container" dict itself?:
```python
container = get_container_env_info()
if container:
    env["container"] = container
```

Reply:
I like that suggestion, let's go with that.